x-pack/plugin/apm-data/src/main/resources/component-templates/logs-apm.app-fallback@ilm.yaml

```diff
@@ -7,5 +7,5 @@ template:
   settings:
     index:
       lifecycle:
-        name: logs-apm.app_logs-default_policy
-        prefer_ilm: false
+        name: logs-apm.app_logs@lifecycle
+        prefer_ilm: true
```

Comment:
Should we keep the old name for 8.x, and update it in 9.0 (if at all)? Do we document the ILM policy names, or just the retention settings?

Reply:
Discussed on Slack, using the old name apparently causes a conflict when the old policy exists. We also discussed that it may be undesirable to have both the apm-data plugin and (older versions of) the APM integration package creating a policy with the same name.

@lahsivjar would be good to know why the conflict occurs. Does that also mean that if someone created `logs-apm.app_logs@lifecycle` before the plugin starts, a conflict will occur? What are the conditions that lead to it?

Reply:
@lahsivjar have you been able to get down to the root of why the name needs to be changed?

Reply:
I think I made some error while testing previously as I can't reproduce the failure -- basically the templates are upgrading without error. Looking at the code, this should be what happens ([ref](https://github.com/elastic/elasticsearch/blob/6cf45366d5ba34f97c40d79439b769e6ad9ab581/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/template/IndexTemplateRegistry.java#L590-L601)). I think I somehow hit the creation CAS operation which prevented the upgrade somehow but I can't reproduce it now -- I will look into this a bit more.